integration_tests/test.sh

```diff
@@ -29,6 +29,8 @@ set -e
 # Run from root of project
```

Comment:
I know not part of this patch in particular, but I'd recommend setting the following:
```
set -eu
set -o pipefail
```

Reply:
`set -u` will require some refactoring, since the un-settedness of `NOSETUP` / `NOSCHEMA` etc are used to give standard default behavior, but allow special cases.